system_cmd.c waf compile error on pi4

https://github.com/monome/norns/blob/e22003b050d27149e4150bc4c53d7b2a0afb7b47/matron/src/system_cmd.c#L91

```
../matron/src/system_cmd.c: In function ‘run_cmd’:
../matron/src/system_cmd.c:91:5: error: ‘strncpy’ output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
     strncpy(cap, capture, len);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~
../matron/src/system_cmd.c:89:18: note: length computed here
     size_t len = strlen(capture);
                  ^~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

Waf: Leaving directory `/home/we/norns/build'
Build failed
 -> task in 'matron' failed with exit status 1 (run with -v to display more information)
```

--- a/matron/src/system_cmd.c
+++ b/matron/src/system_cmd.c
@@ -85,11 +85,13 @@ void *run_cmd(void *cmd) {
     fprintf(stderr, "finished line loop; full buffer: \n");    
     fprintf(stderr, "%s\n", capture);
 #endif
-    
-    size_t len = strlen(capture);
+
+    // strncpy adds null terminator, *unless* src length >= nbytes.
+    // this is atually fine, but new gcc throws warning
+    // so, copy 1 less byte and let strncpy add the null instead of doing it ourselves/
+    size_t len = strlen(capture) - 1;
     char *cap = malloc((len + 1) * sizeof(char));
     strncpy(cap, capture, len);
-    cap[len] = '\0';
     union event_data *ev = event_data_new(EVENT_SYSTEM_CMD);
 
     // this will get freed when the event is handled

--- a/matron/src/system_cmd.c
+++ b/matron/src/system_cmd.c
@@ -86,12 +86,10 @@ void *run_cmd(void *cmd) {
     fprintf(stderr, "%s\n", capture);
 #endif
 
-    // strncpy adds null terminator, *unless* src length >= nbytes.
-    // this is atually fine, but new gcc throws warning
-    // so, copy 1 less byte and let strncpy add the null instead of doing it ourselves/
-    size_t len = strlen(capture) - 1;
-    char *cap = malloc((len + 1) * sizeof(char));
-    strncpy(cap, capture, len);
+    // just use memcpy and include the null terminator
+    size_t len = strlen(capture) + 1;
+    char *cap = malloc(len);
+    memcpy(cap, capture, len);
     union event_data *ev = event_data_new(EVENT_SYSTEM_CMD);
 
     // this will get freed when the event is handled

